openqabot/types/incident.py

```diff
@@ -144,3 +148,53 @@ def contains_package(self, requires: List[str]) -> bool:
                 if package.startswith(req) and package != "kernel-livepatch-tools":
                     return True
         return False
+
+    def has_failures(self, token) -> bool:
+        results = requests.get(
+            QEM_DASHBOARD + "/api/jobs/incident/" + str(self.id), headers=token
+        ).json()
+
+        failed_jobs = self.filter_failures(results)
+        if failed_jobs:
+            log.info("Found %s failed jobs for incident %s:", len(failed_jobs), self.id)
+            if log.isEnabledFor(LOG_DEBUG_LEVEL):
+                list(
+                    map(lambda job: self.log_debug_incident(job["job_id"]), failed_jobs)
+                )
+
+        if not results:
+            raise NoResultsError(
+                "Job setting %s not found for incident" % (str(self.id))
+            )
+
+        # we only need to check if there are any failed jobs
+        return any(failed_jobs)
+
+    def log_debug_incident(self, job_id):
+        log.debug(
+            "Job %s is not marked as acceptable for incident %s",
+            job_id,
+            self.id,
+        )
+
+    def filter_failures(self, results):
+        return [
+            res
+            for res in results
+            if res["status"] not in ("passed", "softfailed")
```

Comment:
Sorry if this was already asked, but would using `if res["status"] not in ("passed", "softfailed")` treat running/not-completed jobs as failures, and filter more than what is intended?

Reply:
This is addresed in https://github.com/openSUSE/qem-bot/pull/154#discussion_r1474042954 since we're querying smelt, it makes sense to use that data instead of anything else

Reply:
That's is correct, but since aggregates are scheduled late in the evening (European time) I imagine the impact is not huge. And right now if a new RR comes out after aggregates are scheduled, they also wait another day, so we are enlarging that window a little bit (depending on how much time incidents need to finish)